dts/bindings/timer/st,stm32-lptim.yaml

```diff
@@ -10,3 +10,32 @@ include:
     property-blocklist:
         - st,prescaler
         - st,countermode
+
+properties:
```

Comment:
The prescaler value is computed and set at runtime, it's part of software state, not hardware configuration.  This wants to be a kconfig, doesn't it?

Reply:
This prescaler is actually one register of the timer, so to me this is indeed a hardware configuration (we do have the same on similar timer IPs like the ones used for PWM and we also configure them using device tree).
What we're doing here is really an hardware configuration. But indeed, the actual count per tick will then be computed through SYS_CLOCK_TICKS_PER_SEC, which is Kconfig based.
Does it answer your point ?

Reply:
You're computing a value at runtime and setting it in the register to change device state based on how the user of the app (and not the owner of the platform code) wants the hardware to behave.  To me that's 100% kconfig's job.

Devicetree is best used to express immutable facts about hardware ("this SOC has an I2C controller at this address",  "this board puts the first SPI clock on pin 41", "this CPU variant has the FPU enabled").  Kconfig is for things that the application wants to tune ("enable this feature", "include this driver", "reserve this many items for this object cache").

There's absolutely some gray area between the two where there's reasonable argument about which to use.  And there are for sure places still in the code where we're using one when we should be using the other (CONFIG_SYS_CLOCK_CYCLES_PER_SEC is actually one of them -- that's a hardware feature, not an app tunable).

But this particular situation seems pretty clear to me.   You support this prescaler so the **application** can make a tradeoff decision between timing precision and idle power draw.  If you as the platform owner can't make the right decision always, it belongs in the app's hands, which means kconfig.